src/v/raft/consensus_utils.cc

```diff
@@ -193,6 +195,27 @@ model::record_batch make_ghost_batch(
     return batch;
 }
 
+/**
+ * makes multiple ghost batches required to fill the gap in a way that max batch
+ * size (max of int32_t) is not exceeded
+ */
+std::vector<model::record_batch> make_ghost_batches(
+  model::offset start_offset, model::offset end_offset, model::term_id term) {
+    std::vector<model::record_batch> batches;
+    while (start_offset < end_offset) {
+        auto int_end_offset = std::clamp(
```

Comment:
I think you should make this a function with some tests. 

But great catch.

Reply:
this is a function (whole one) it is covered with tests in next commit of this PR.